tests/src/rules/no-default-export.js

```diff
@@ -0,0 +1,123 @@
+import { test } from '../utils'
+
+import { RuleTester } from 'eslint'
+
+const ruleTester = new RuleTester()
+    , rule = require('rules/no-default-export')
+
+ruleTester.run('no-default-export', rule, {
+  valid: [
+    test({
+      code: `
+        export const foo = 'foo';
+        export const bar = 'bar';`,
+      }),
+    test({
+      code: `
+        export const foo = 'foo';
+        export function bar() {};`,
+      }),
+    test({
+      code: `
+        export const foo = 'foo';`,
+      }),
+    test({
+      code: `
+        const foo = 'foo';
+        export { foo };`,
+      }),
+    test({
+      code: `
+        export { foo, bar }`,
+      }),
+    test({
+      code: `
+        export const { foo, bar } = item;`,
+      }),
+    test({
+      code: `
+        export const { foo, bar: baz } = item;`,
+      }),
+    test({
+      code: `
+        export const { foo: { bar, baz } } = item;`,
+      }),
+    test({
+      code: `
+        export const foo = item;
+        export { item };`,
+      }),
+    test({
+      code: `
+        export * from './foo';`,
+      }),
+    test({
+      code: `
+        export const { foo } = { foo: "bar" };`,
+      }),
+    test({
+      code: `
+        export const { foo: { bar } } = { foo: { bar: "baz" } };`,
+      }),
+    test({
+      code: 'export { a, b } from "foo.js"',
+      parser: 'babel-eslint',
+      }),
+
+    // no exports at all
+    test({
+      code: `
+        import * as foo from './foo';`,
+      }),
+
+    test({
+      code: `export type UserId = number;`,
+      parser: 'babel-eslint',
+      }),
+
+    // ...SYNTAX_CASES,
+  ],
```

Comment:
Can you add a test for something like `import defaultImport from 'module-from-npm'`? This should be legal, since we can’t control how other libraries are structured.

Reply:
I didn't add a test for it, since it only checks `export` and `export ... from`. I can add a test to clarify this, though.